mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -281,13 +281,22 @@ def upload_trace_data(self, trace_data: str) -> None:
                     get_credentials=lambda artifact_paths: [
                         self._get_upload_trace_data_cred_info()
                     ],
+                    is_sync=True,
                 )
             elif (
                 cred.type == ArtifactCredentialType.AWS_PRESIGNED_URL
                 or cred.type == ArtifactCredentialType.GCP_SIGNED_URL
             ):
                 self._signed_url_upload_file(cred, temp_file)
 
+    def _get_upload_trace_data_cred_info(self):
```

Comment:
This is not directly related to the exact issue, but this method has been missing from DatabricksArtifactRepository while it's used